File: 1022-sum-of-root-to-leaf-binary-numbers/1022-sum-of-root-to-leaf-binary-numbers.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int sumRootToLeaf(TreeNode* root) {
        int res = 0;
        solve(root, 0, res);
        return res;
    }
private:
    void solve(TreeNode* root, int sum, int& res)
    {
        if(!root)
        {
            return;
        }
        if(!root->left && !root->right)
        {
            res += sum * 2 + root->val;
            return; 
        }
        solve(root->left, sum * 2 + root->val, res);
        solve(root->right, sum * 2 + root->val, res);
    }
};

File: 1022-sum-of-root-to-leaf-binary-numbers/1022-sum-of-root-to-leaf-binary-numbers.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def sumRootToLeaf(self, root: Optional[TreeNode]) -> int:
        paths = []
        def solve(root: Optional[TreeNode], tot: int) -> None:
            if not root:
                return
            if (not root.left) and (not root.right):
                paths.append(tot * 2 + root.val)
                return
            solve(root.left, tot * 2 + root.val)
            solve(root.right, tot * 2 + root.val)
        solve(root, 0)
        print(paths)
        return sum(paths)
            